### ISSUE TITLE
Stream Has No Promise Result

### ISSUE THREAD

**Issue by shellscape** [knex/knex#5922]
The docs for [Streams](https://knexjs.org/#Interfaces-Streams) contain a code example that doesn't work:

```js
// Use as a promise:
const stream = knex.select('*').from('users')
  .where(knex.raw('id = ?', [1]))
  .stream(function(stream) {
    stream.pipe(writableStream);
  })
  .then(function() { // ... })
  .catch(function(e) { console.error(e); });
```

The result of the `.stream(...)` call is a `Transform` stream, not a `Promise`.

**Comment by mercmobily** [comment]
You're correct that the documentation example labelled "Use as a promise" can be confusing. However, the behaviour shown is actually technically correct - when you pass a callback function to `.stream()`, it returns a promise rather than a stream. The implementation in lib/execution/runner.js:111-118 shows that if a handler function is provided, the stream is passed to that handler and a promise is returned. Without a handler, `.stream()` returns the Transform stream directly.

The source of confusion is that the example shows both behaviours without clearly explaining when each applies. The documentation at line 116 states "If called with a callback, the callback is passed the stream and a promise is returned. Otherwise, the readable stream is returned" - which is accurate - but the "Use as a promise" example (lines 140-155) could be more clearly labelled to indicate it only returns a promise because a callback function is passed to `.stream()`.

The additional issues raised in the comments are valid documentation problems that need addressing:

The link to substack's stream handbook (docs/src/guide/interfaces.md:110) points to an archived repository. This should be updated to point to the official Node.js streams documentation (https://nodejs.org/api/stream.html) or another current resource about streams.

The reference to pg-query-stream (docs/src/guide/interfaces.md:110) mentions an archived module. While the module still functions and Knex continues to use it as an optional dependency (package.json lists pg-query-stream@^4.2.4), the documentation should note that the module is archived and no longer actively maintained. The module is still required for PostgreSQL streaming, so the functionality works, but users should be aware of its maintenance status.

A documentation improvement PR should address these points: (1) Clarify the dual behaviour of `.stream()` - make it explicit that passing a callback returns a promise, while calling it without arguments returns a stream, (2) Replace or supplement the substack stream handbook link with current Node.js stream documentation, (3) Add a note that pg-query-stream, though still functional and used by Knex, is an archived project, and (4) Consider adding a clearer example showing the difference between `.stream()` and `.stream(callback)`.

Related issues:
- #5903 - Streaming example (request for better streaming documentation/recipes)
- #2037 - Original issue about streaming that was moved to documentation repository

**Comment by shellscape** [comment]
Hey better late (3 years) than never I suppose. Have long since moved on from knex. 👋🏼

**Comment by mercmobily** [comment]
I am reopening the issue since we still need to update the documentation. Sorry to see you go...!
